src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/DisableRuntimeMarshallingAttribute.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*=============================================================================
+**
+**
+**
+** Purpose: Mark up the program to indicate various legacy or new opt-in behaviors.
+**
+**
+=============================================================================*/
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Disables the built-in runtime managed/unmanaged marshalling subsystem for
+    /// P/Invokes, Delegate types, and unmanaged function pointer invocations.
+    /// </summary>
+    /// <remarks>
+    /// The built-in marshalling subsystem has some behaviors that cannot be changed due to
+    /// backward-compatibility requirements. This attribute allows disabling the built-in
+    /// subsystem and instead uses the following rules for P/Invokes, Delegates,
+    /// and unmanaged function pointer invocations:
+    ///
+    /// - All value types that do not contain reference type fields recursively (<c>unmanaged</c> in C#) are blittable
+    /// - Value types that recursively have any fields that are not enums and have <c>[StructLayout(LayoutKind.Auto)]</c> are disallowed from interop.
```

Comment:
I would omit the mention of enums from the public docs. It is a uninteresting low-level IL corner case and mentioning it here reduces clarity.